src/Pos/Client/Txp/History.hs

```diff
@@ -120,10 +122,11 @@ getRelatedTxs addr txs = fmap DL.toList $
     handleRelatedTx (isOutgoing, isToItself) (tx, txId, dist) = do
         applyTxToUtxo (WithHash tx txId) dist
         ether $ identity %= filterUtxoByAddr addr
+        inputs <- (map toaOut . catMaybes) <$> mapM utxoGet (toList $ _txInputs tx)
```

Comment:
The answer to your first question is precedence. `($)` has the lowest precedence of any operator. If you replace the parenthesis with a `($)`, then the expression actually becomes

`((map toaOut . catMaybes) <$> mapM utxoGet) $ toList $ _txInputs tx`

which probably causes the type error. As for the second question, I can't build the code right now, but if you keep the RHS of `(<$>)` as it is, I'm fairly sure you can remove those parens without a problem.